x-pack/agent/pkg/agent/application/fleet_gateway.go

```diff
@@ -5,8 +5,12 @@
 package application
 
 import (
+	"sync"
 	"time"
 
+	"github.com/pkg/errors"
```

Comment:
can you use custom errors package and do a `errors.TypeNetwork` type with `errors.M(errors.MetaKeyURI, "URI")`?